programs/marginfi/src/instructions/marginfi_account/repay.rs

```diff
@@ -71,6 +71,11 @@ pub fn lending_account_repay<'info>(
 
         amount
     };
+    marginfi_account.last_update = if repay_all {
```

Comment:
here's it's probably easier to use clock time always rather than the balance conditionally